clients/rospy/src/rospy/client.py

```diff
@@ -300,7 +296,13 @@ def init_node(name, argv=None, anonymous=False, log_level=None, disable_rostime=
         # hostname as that is not guaranteed to be a legal ROS name
         name = "%s_%s_%s"%(name, os.getpid(), int(time.time()*1000))
 
-    resolved_node_name = rospy.names.resolve_name(name)
+    # check for legal base name once all changes have been made to the name
+    if not rosgraph.names.is_legal_base_name(name):
+        import warnings
+        warnings.warn("'%s' is not a legal ROS base name. This may cause problems with other ROS tools. This will be an error in future ROS releases."%name, stacklevel=2)
```

Comment:
After thinking about this a bit more I am not sure if I would escalate it to be an error in the future. If the user wants to ignore the recommendation and warning we don't have to block him from doing it.

Therefore I would propose to remove `This will be an error in future ROS releases.` for now.